sqlite3/src/lib.rs

```diff
@@ -1646,3 +1674,151 @@ fn sqlite3_safety_check_sick_or_ok(db: &sqlite3Inner) -> bool {
         }
     }
 }
+
+#[no_mangle]
+pub unsafe extern "C" fn sqlite3_table_column_metadata(
+    db: *mut sqlite3,
+    z_db_name: *const ffi::c_char,
+    z_table_name: *const ffi::c_char,
+    z_column_name: *const ffi::c_char,
+    pz_data_type: *mut *const ffi::c_char,
+    pz_coll_seq: *mut *const ffi::c_char,
+    p_not_null: *mut ffi::c_int,
+    p_primary_key: *mut ffi::c_int,
+    p_autoinc: *mut ffi::c_int,
+) -> ffi::c_int {
+    trace!("sqlite3_table_column_metadata");
+
+    let mut rc = SQLITE_OK;
+    let mut z_data_type: *const ffi::c_char = std::ptr::null();
+    let mut z_coll_seq: *const ffi::c_char = std::ptr::null();
+    let mut not_null = 0;
+    let mut primary_key = 0;
+    let mut autoinc = 0;
+
+    // Safety checks
+    if db.is_null() || z_table_name.is_null() {
+        return SQLITE_MISUSE;
+    }
+
+    let db_inner = &mut *db;
+    let db_inner = match db_inner.inner.lock() {
```

Comment:
I think we can just unwrap here. Lock can be poisoned if code under lock panicked somewhere - but in this case most probably program already in the invalid state and we can panic here "safely" too.